SurfaceTopography/Generic/ScaleDependentStatistics.py

```diff
@@ -121,19 +127,29 @@ def scale_dependent_statistical_property(self, func, n=1, scale_factor=None, dis
                 iter(scale_factor)
             if distance is not None:
                 iter(distance)
-            return distance, [func(_d) for _d in d]
+            retvals = np.array([func(_d) for _d in d])
         except TypeError:
+            # If there is a single distance, the reliability analysis is skipped
             return distance, func(d)
     else:
         try:
             if scale_factor is not None:
                 iter(scale_factor)
             if distance is not None:
                 iter(distance)
-            return distance, [func(dx, dy) for dx, dy in zip(*d)]
+            retvals = np.array([func(dx, dy) for dx, dy in zip(*d)])
         except TypeError:
+            # If there is a single distance, the reliability analysis is skipped
             return distance, func(*d)
 
+    short_cutoff = self.short_reliability_cutoff() if reliable else None
+    if short_cutoff is not None:
+        mask = distance > short_cutoff / 2
```

Comment:
Thanks for clarifying the docstring. But then I am a bit doubtful about this. 
For curvatures, I would cutoff data with stencil lengths smaller then the stencil length that we used for the peak curvature analysis. 

For other orders of derivatives I would make the cutoff based on the pixel size, intuitively. Any thoughts ?

Reply:
You are right. This was taken verbatim from `autocorrelation` where this criterion is correct.